server/level_loader.cpp

```diff
@@ -23,69 +23,67 @@ void initPlayerwall(const FlatBuffGenerated::PlayerWall *pw)
 	gameState.level->playerwalls.back()->body = staticBody;
 }
 
-void initStone(Stone *s, const FlatBuffGenerated::Stone *dfstone)
-{
-	b2BodyDef myBodyDef;
-	myBodyDef.type = b2_staticBody;
-	myBodyDef.position.Set(dfstone->pos()->x(), dfstone->pos()->y());
-	myBodyDef.angle = -dfstone->rotation();
-	s->body = gameState.b2world->CreateBody(&myBodyDef);
-	b2CircleShape circleShape;
-	circleShape.m_radius = dfstone->radius();
-
-	b2FixtureDef fixtureDef;
-	fixtureDef.shape = &circleShape;
-	fixtureDef.density = 1;
-	s->body->CreateFixture(&fixtureDef);
-	s->body->SetUserData(s);
-}
-
-void initHidingSpot(HidingSpot *b, const FlatBuffGenerated::HidingSpot *dfhspot)
-{
-	b2BodyDef myBodyDef;
-	myBodyDef.type = b2_staticBody;
-	myBodyDef.position.Set(dfhspot->pos()->x(), dfhspot->pos()->y());
-	myBodyDef.angle = -dfhspot->rotation();
-	b->body = gameState.b2world->CreateBody(&myBodyDef);
-	b2CircleShape circleShape;
-	circleShape.m_radius = dfhspot->radius();
-
-	b2FixtureDef fixtureDef;
-	fixtureDef.shape = &circleShape;
-	fixtureDef.density = 1;
-	fixtureDef.filter.categoryBits = 0x0002;
-	b->body->CreateFixture(&fixtureDef);
-	b->body->SetUserData(b);
-}
-
 flatbuffers::Offset<FlatBuffGenerated::Level> serializeLevel(flatbuffers::FlatBufferBuilder &builder)
 {
+	//tileinfo
+	std::vector<flatbuffers::Offset<FlatBuffGenerated::Tileinfo>> tileinfoOffsets;
+	for (auto &ts : gameState.level->tileinfo) {
+		auto name = builder.CreateString(ts->name);
+		auto offset = FlatBuffGenerated::CreateTileinfo(builder, ts->gid, name);
+		tileinfoOffsets.push_back(offset);
+	}
+	auto tilesets = builder.CreateVector(tileinfoOffsets);
+
+	// objects
+	std::vector<flatbuffers::Offset<FlatBuffGenerated::Object>> objectOffsets;
+	for (auto &o : gameState.level->objects) {
+		FlatBuffGenerated::Vec2 pos(o->pos.x, o->pos.y);
+		auto hspotname = builder.CreateString(o->hspotname);
+		auto offset = FlatBuffGenerated::CreateObject(builder, &pos, o->rotation, o->gid, o->hspot, hspotname);
+		objectOffsets.push_back(offset);
+	}
+	auto objects = builder.CreateVector(objectOffsets);
+
+	// decoration
+	std::vector<flatbuffers::Offset<FlatBuffGenerated::Decoration>> decorationOffsets;
```

Comment:
ugly and repetitive code that we can do nothing about without using macros because flatbuffers are mentally challenged

Reply:
We could try pushing it into a function with an additional parameter and iffing, but I think it would be much less readable and much more sore for the eyes. The copy-pasted operation isn't that long and that similar for this to make sense, imo.